ros2nix/ros2nix.py

```diff
@@ -247,6 +250,13 @@ def ros2nix(args):
         "The fetch function and its parameters are determined from the local git work tree. "
         "sourceRoot attribute is set if needed and not overridden by --source-root.",
     )
+
+    parser.add_argument(
+        "--use-package-git-hash",
+        action="store_true",
+        help="When using --fetch, use the git hash of the package sub-directory instead of the one of the upstream repo."
+        "This will lead to longer generation time and multiple source checkouts when building but will safe rebuilds of packages that have not changed."
+    )
```

Comment:
What about changing the option name and description like this?

```suggestion
    parser.add_argument(
        "--use-per-package-src",
        action="store_true",
        help="When using --fetch, fetch only the package sub-directory instead of the whole repo. "
        "For repos with multiple packages, this will avoid rebuilds of unchanged packages at the cost of longer generation time."
    )
```

I think you wrote it before using sparse checkout and it's not accurate now.